### ISSUE TITLE
evaluate ETH DMA design

### ISSUE THREAD

**Issue by jordens** [quartiq/stabilizer#84]
Use `VolatileCell` along the lines of [stb32-usbd](https://github.com/stm32-rs/stm32-usbd/blob/master/src/endpoint_memory.rs) and check what evolves around best practices of [DMA/uninit](https://zulip-archive.rust-lang.org/136281tlangwgunsafecodeguidelines/99575UninitdataandDMA.html), (MaybeUninit, compiler_fence).

**Comment by ryan-summers** [comment]
@jordens I believe this can be closed since we're using [`stm32h7-ethernet`](https://github.com/richardeoin/stm32h7-ethernet)

**Comment by jordens** [comment]
I don't think so. `stm32h7-ethernet` doesn't mark the descriptors as `VolatileCells` either. It doesn't differ from my old code in that respect AFAICT.

**Comment by ryan-summers** [comment]
What I was trying to imply is that this issue isn't necessary suitable for this repository anymore, but rather the linked `stm32h7x-ethernet` crate since we don't implement any ethernet DMA functionality here.

**Comment by jordens** [comment]
Oh yes. Right. Somebody with a sound understanding of this should move it over to `stm32h7-ethernet` if appropriate.
